openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -177,6 +177,14 @@ def get_response_json(self, query_params: Dict):
             self.endpoint, self.retries, query_params, headers=self.headers
         )
 
+    def get_detail_json(self, endpoint: str, query_params: Dict | None = None):
```

Comment:
Since this is almost exactly the same as `get_response_json`, do you think we could just use that instead and pass in `{}` if necessary in the operational code?

Reply:
Ah, I'm noticing now that this allows the specification of an endpoint! I also have a use for this in #583 - I'll see if I can still fold this into the `get_response_json` function in my PR!

Reply:
Here's how I plan on modifying it on my branch:

```diff
diff --git a/openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py b/openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py
index 8d9d2efc..adf19295 100644
--- a/openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py
+++ b/openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py
@@ -315,13 +315,16 @@ class ProviderDataIngester(ABC):
 
         return batch, should_continue
 
-    def get_response_json(self, query_params: dict):
+    def get_response_json(self, query_params: dict, endpoint: str | None = None):
         """
         Make the actual API requests needed to ingest a batch. This can be overridden
         in order to support APIs that require multiple requests, for example.
         """
         return self.delayed_requester.get_response_json(
-            self.endpoint, self.retries, query_params, headers=self.headers
+            endpoint or self.endpoint,
+            self.retries,
+            query_params,
+            headers=self.headers,
         )
 
     def get_should_continue(self, response_json):
```

Reply:
This is great, thank you @AetherUnbound !